src/extension_instance.ml

```diff
@@ -153,18 +182,41 @@ end
 let make () =
   let sandbox = Sandbox.Global in
   let sandbox_info = Sandbox_info.make sandbox in
+  let documentation_server_info = Documentation_server_info.make () in
   let ast_editor_state = Ast_editor_state.make () in
   { sandbox
   ; lsp_client = None
   ; sandbox_info
+  ; documentation_server_info
   ; repl = None
   ; ocaml_version = None
   ; ast_editor_state
+  ; documentation_server = None
   }
 
+let document_server_on = "ocaml.documentation-server-on"
+
+let stop_documentation_server t =
+  Option.value_map ~default:(Promise.resolve ()) t.documentation_server
+    ~f:(fun server ->
+      let open Promise.Syntax in
+      let+ (_ : (unit, Promise.error) result) =
+        Documentation_server.stop server
+      in
+      t.documentation_server <- None;
+      let (_ : Ojs.t option Promise.t) =
+        Vscode.Commands.executeCommand ~command:"setContext"
+          ~args:[ Ojs.string_to_js document_server_on; Ojs.bool_to_js false ]
+      in
+      StatusBarItem.hide t.documentation_server_info)
```

Comment:
Why don't we dispose of the status bar item? If we don't, I'm not sure that reloading the plugin is going to be reliable.